Library/Homebrew/dev-cmd/edit.rb

```diff
@@ -59,19 +59,28 @@ def edit
           "#{path} doesn't exist on disk."
         end
 
-        message = if args.cask?
-          <<~EOS
-            #{not_exist_message}
-            Run #{Formatter.identifier("brew create --cask --set-name #{path.basename(".rb")} $URL")} \
-            to create a new cask!
-          EOS
+        suggestion_message = if args.cask?
+          if !CoreCaskTap.instance.installed? && Homebrew::API::Cask.all_casks.key?(path.basename(".rb").to_s)
```

Comment:
Would it also be worth checking that the path is associated with the core tap that is missing here?

We already check the path with `#fnmatch?` below in the same file and maybe those checks can be reused somehow. It might be going overboard though. More than half of this file is just warning messages and error handling now.

Reply:
Which named arg formats should the command be expected to handle? I assume "org/tap/fname", "org/tap/cname", "fname", and "cname"; any others?

Reply:
Those should all return the same paths when `NamedArgs#to_paths` is called from what I understand.

```rb
brew(main):006:0> Homebrew::CLI::NamedArgs.new("jrnl", "Homebrew/core/jrnl", "/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/jrnl.rb")
=> ["jrnl", "Homebrew/core/jrnl", "/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/jrnl.rb"]
brew(main):007:0> args = _                                                                          => ["jrnl", "Homebrew/core/jrnl", "/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/jrnl.rb"]
brew(main):008:0> args.to_paths
=> [#<Pathname:/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/jrnl.rb>]
```